mantidproject/mantid
# Engineering GUI implement new model for EnggVanadiumCorrections

A model for generating and caching the vanadium corrections workspaces (calculated by EnggVanadiumCorrections and used for calibration and focusing) was implemented and integrated into the main presenter of the ED GUI. The intended behaviour was always to cache the workspaces so that we don't need to run EnggVanadiumCorrections (which is slow) for every calibration or focus, but the caching didn't actually appear to happen. The new model takes care of the algorithm and the cache, and also gets unit tests on the behaviour.

**To test:**

There's some sample data [here](https://github.com/mantidproject/mantid/files/1967047/EnggCalibrateAndFocus.zip)

Open up the GUI and go to the Calibration tab. Under **Make new calibration**, select the 280329 run for Vanadium and 280336 for Calibration sample. Click **Calibrate**, and go and make a coffee for a few minutes. Once this is complete, make sure that the two workspaces were saved out (you can see where they were saved in the Calibration field in the Settings tab).

Now go to the Focus tab and select the 281188 run and click **Focus**. Make sure that EnggVanadiumCorrections isn't run by looking in the Algorithm Progress window.

Repeat any of the above steps, and EnggVanadiumCorrections still shouldn't be run

Back in the Settings tab, tick **Force recalculate all existing Vanadium files**. Repeat one of the above steps and make sure EnggVanadiumCorrections is run again, and the files are overwritten

Fixes #17424

**Release Notes** 

Doesn't need to go in the release notes, as it's mostly a refactoring change, and I don't think the scientists were aware that the cache didn't work in the first place

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
